Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Search.hs

```diff
@@ -148,3 +148,12 @@ buildRoutePoints req = do
       >>= (.fulfillmentTags)
       & fromMaybeM (InvalidRequest "Missing Tags")
   return $ getTagV2 "route_info" "route_points" tagGroups >>= decode . encodeUtf8 >>= Just
+
+buildMultipleRoutesTag :: MonadFlow m => Spec.SearchReqMessage -> m (Maybe [Maps.RouteInfo])
+buildMultipleRoutesTag req = do
+  tagGroups <-
+    req.searchReqMessageIntent
+      >>= (.intentFulfillment)
+      >>= (.fulfillmentTags)
+      & fromMaybeM (InvalidRequest "Missing Tags")
```

Comment:
Handle backward compatibility when BPP is released first then BAP will not send multiple routes at that time so don't throw error, handle the flow with single route in that case as fallback